src/panel/components/CodeHighlight.tsx

```diff
@@ -9,19 +15,40 @@ export const CodeHighlight: FC<
     language: PrismLanguage;
   } & ComponentPropsWithoutRef<typeof StyledCodeBlock>
 > = ({ code, language, ...props }) => {
+  const [visible, setVisibility] = useState(false);
+  const [copy, setCopied] = useState(false);
+
+  const handleClick = async () => {
+    const text = document.getElementsByClassName("language")[0].textContent;
+    if (text) {
+      try {
+        await navigator.clipboard.writeText(text);
+        setCopied(true);
```

Comment:
Oversight on my part - we probably want the "copied" text to last for N seconds following the last click (which won't work with a useEffect).

To get around this, we could either store the copied state in an object (allowing us to know when a subsequent copy has taken place thanks to the referential update)

```tsx
const copied = useState({ state: true });

useEffect(() => {
 // ...
}, [copied])
```

Or if you prefer, we could implement a debounced "clearCopiedState" function

```tsx
// Somewhere in utils
export const debounce = <T extends Function>(fn: T, delay: number): T => {
  let timeout;
  (...args: Parameters<T>) => {
    if (timeout) {
      clearTimeout(timeout);
    }

    timeout = setTimeout(() => fn(...args), delay);
  }
});

// In component
const clearCopiedState = useCallback(() => debounce(() => setCopied(false), 1000), []);

// ...
setCopied(true);
clearCopiedState();
```